core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -2852,16 +2881,60 @@ wasi_ssp_sock_addr_local(
 #if !defined(WASMTIME_SSP_STATIC_CURFDS)
     struct fd_table *curfds,
 #endif
-    __wasi_fd_t fd, uint8 *buf, __wasi_size_t buf_len)
+    __wasi_fd_t fd, __wasi_addr_t *addr)
 {
     struct fd_object *fo;
+    char buf[64];
+    __wasi_ip_port_t port;
+    uint8_t is_ipv4;
+    int ret;
+    uint8_t inet_pton_dest[16];
+
     __wasi_errno_t error =
         fd_object_get(curfds, &fo, fd, __WASI_RIGHT_SOCK_ADDR_LOCAL, 0);
     if (error != __WASI_ESUCCESS)
         return error;
 
+    ret = os_socket_addr_local(fd_number(fo), buf, sizeof(buf) / sizeof(buf[0]),
+                               &port, &is_ipv4);
     fd_object_release(fo);
-    return __WASI_ENOSYS;
+    if (ret != BHT_OK) {
+        error = convert_errno(errno);
+    }
+
+    if (BHT_OK != os_socket_inet_pton(is_ipv4, buf, inet_pton_dest)) {
```

Comment:
wondering why convert a network order binary to presentation in `os_socket_addr_local()` and convert it back here?

Reply:
Yeah, you're right; I guess I was confused by the way it's done in `bind()` - I think we could pass binary representation there as well and that would simplify the code.